### PR TITLE
Simplify GenericParamList nesting

### PR DESCRIPTION
**Pull request by slavapestov**
Follow-up to https://github.com/apple/swift/pull/21226.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: include/swift/Serialization/ModuleFile.h [hunk 1]
```diff
@@ -562,8 +562,7 @@ class ModuleFile
   ///
   /// If the record at the cursor is not a generic param list, returns null
   /// without moving the cursor.
-  GenericParamList *maybeReadGenericParams(DeclContext *DC,
-                                     GenericParamList *outerParams = nullptr);
+  GenericParamList *maybeReadGenericParams(DeclContext *DC);
```
Comment (DougGregor) [inline include/swift/Serialization/ModuleFile.h]: Could you bump the module file version? Even though you didn't change the code, the way we represent generic parameter lists is so different that there's zero compatibility between module files before/after.

**Pull request merged**